### PR TITLE
RHEL8

### PR DESCRIPTION
**Pull request by gombasg**
RHEL 8 support for AII. Only the creation of /etc/systemd/system/system-update.target.wants is really needed, the rest is mostly cosmetic.

### CODE REVIEW

File: aii-ks/src/main/perl/ks.pm [hunk 1]
```diff
@@ -984,12 +988,16 @@ sub ksinstall_rpm
...
+    my $tree = $config->getElement(KS)->getTree;
+    my $version = get_anaconda_version($tree);
+
     # DISABLED_REPOS doesn't exist in 13.1
     my $disabled = [];
     if ( $config->elementExists(DISABLED_REPOS) ) {
         $disabled = $config->getElement(DISABLED_REPOS)->getTree();
     }
-    my $cmd = "yum -c /tmp/aii/yum/yum.conf -y install ";
+    my $packager = $version >= ANACONDA_VERSION_EL_8_0 ? "dnf" : "yum";
```
Comment (stdweird) [inline aii-ks/src/main/perl/ks.pm]: i don't like this one. it's confusing with the yum.conf, and yum as command is fully supported.

File: aii-ks/src/main/perl/ks.pm [hunk 2]
```diff
@@ -1684,6 +1693,7 @@ EOF_reboot_unit
     ln -sf /etc/rc.d/init.d/ks-post-reboot /system-update
 
     # The documentation recommends creating the .wants symlink directly instead of using [Install] and 'systemctl enable'
+    mkdir -p /etc/systemd/system/system-update.target.wants
```
Comment (stdweird) [inline aii-ks/src/main/perl/ks.pm]: i ran into this one as well, was making a PR too :wink:

**Comment by stdweird** [comment]
@gombasg i'm working on a PR too. i have another bunch of code refactor to allow the configuration of the install repos based on the configured repositories in the profile (instead of the current string). we need appstream as addiitonal repo for early authconfig support.

**Comment by gombasg** [comment]
@stdweird You mean converting `/system/aii/osinstall/ks/repo` from a list-of-strings to a list-of-dicts? That would be nice :-) However, the list of repositories used for Kickstart should remain distinct from the repositories used by spma, populating `/system/aii/osinstall/ks/repo` from the spma repositories would not work for us.

If you're working on that, it would also be nice to make `/system/aii/osinstall/ks/installtype` a dictionary, too, and add support for using `liveimg` :-)

**Comment by stdweird** [comment]
@gombasg no, i'm was going to change the code a bit so you can pass a "magic" name (like `@appstream` to the `/system/aii/osinstall/ks/repo` and it would add all spma repos matching that name. i'm not touching any of the clunky lists that should be dicts :smile:

**Comment by stdweird** [comment]
test this please

**Pull request merged**